src/vcBPA.cpp

```diff
@@ -831,7 +846,11 @@ uint32_t vcBPA_GridGeneratorThread(void *pDataPtr)
 
   vcBPAGrid *pGrid = nullptr;
   int i = 0;
-  while (vcBPA_GetGrid(pData->pManifold, pData->pNewModel, pData->standardContent, &pGrid, false) && pData->running)
+
+  vdkPointCloudHeader header;
+  vdkPointCloud_GetHeader(pData->pNewModel, &header);
```

Comment:
If this somehow fails, the header is going to be garbage data and the attributes will likely trigger a crash.

Reply:
Yeah, sorry- knowing internal implementation broke this. I know that function can't fail in its current form and missed this. I'll add the checks now.